linux/deploy_vm/deploy_vm_from_ova.yml

```diff
@@ -56,15 +56,25 @@
       vars:
         snapshot_name: "{{ base_snapshot_for_reconfig }}"
 
-    - name: "Power on VM"
-      include_tasks: ../../common/vm_set_power_state.yml
-      vars:
-        vm_power_state_set: 'powered-on'
-
     - name: "Collect OVA configs"
       include_tasks: "../../common/collect_ovf_vm_config.yml"
       vars:
         ovf_vm_hardware_config_path: "{{ current_test_log_folder }}"
+    
+    - name: "Get VM info"
+      include_tasks: ../../common/vm_get_vm_info.yml
+
+    - name: "Check toolsVersion of open-vm-tools in OVA after deploy VM"
+      block:
+        - name: "Display toolsVersion of open-vm-tools in OVA"
+          debug: var=ovf_vm_vmtools_info['Config Tools Version']
+        
+        - name: "Check toolsVersion of open-vm-tools in OVA is expected"
+          ansible.builtin.assert:
+            that:
+              - ovf_vm_vmtools_info['Config Tools Version'] in ['2147483647', vm_extra_config['guestinfo.vmtools.versionNumber']]
+            fail_msg: "toolsVersion of open-vm-tools in OVA is not expected"
+      when: ovf_vm_vmtools_info['Guest Tools Version'] is match("build-")
```

Comment:
You can put this checking to the end of `common/collect_ovf_vm_config.yml`, so that both linux and windows OVA can be checked.

In the assertion, you need to firstly check ovf_vm_vmtools_info['Config Tools Version'] or vm_extra_config['guestinfo.vmtools.versionNumber'] is not empty. And
please change the fail message be more specific. For example,
```
fail_msg: "The ddb.toolsVersion in OVA template is {{ ovf_vm_vmtools_info['Config Tools Version'] }}, not as expected '2147483647' or {{ vm_extra_config['guestinfo.vmtools.versionNumber']] }}.
```

The when condition is not needed

Reply:
Change it. Thanks